engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/PassManager.scala

```diff
@@ -100,13 +121,36 @@ class PassManager(
 
         pass match {
           case miniFactory: MiniPassFactory =>
-            val miniPass = miniFactory.createForModuleCompilation(newContext)
-            MiniIRPass.compile(classOf[Module], intermediateIR, miniPass)
+            logger.trace(
+              "  mini collected: {}",
+              pass
+            )
+            val combiningPreventedBy = pendingMiniPasses.find { p =>
+              p.asInstanceOf[IRPass].invalidatedPasses.contains(miniFactory)
+            }
+            val irForRemainingMiniPasses = if (combiningPreventedBy.isDefined) {
+              logger.trace(
+                "  pass {} forces flush before {}",
+                combiningPreventedBy.orNull,
+                miniFactory
+              )
+              flushMiniPass(intermediateIR)
+            } else {
+              intermediateIR
+            }
```

Comment:
With the 2736a76 changes the `PassManager` combines _subsequent mini passes_ unless the already collected passes prevent such combining by declaring newly collected pass to be _invalidated_ by their changes. Imagine there is `m.enso` file:
```ruby
import Standard.Base.Runtime.Ref.Ref

main =
    r = Ref.new "foo"
    r.modify (_+"boo")
    r.get
```
then the processing of the `m.enso` file logs following info:
```
runPassesOnModule[m@AFTER_IMPORT_RESOLUTION]
   mega running: MethodDefinitions
   mini collected: org.enso.compiler.pass.desugar.SectionsToBinOp@784abd3e
   mini collected: OperatorToFunction
   mini collected: LambdaShorthandToLambda
   pass OperatorToFunction forces flush before LambdaShorthandToLambda
   flushing pending mini pass: org.enso.compiler.pass.desugar.SectionsToBinOp$Mini:org.enso.compiler.pass.desugar.OperatorToFunctionMini
   flushing pending mini pass: org.enso.compiler.pass.desugar.LambdaShorthandToLambdaMini
   mega running: ImportSymbolAnalysis
   ...
```
e.g. **three mini passes** are collected in this `PassGroup`. Two of them are _"flushed"_ together by a single IR traversal. The 3rd one is _"flushed"_ separately before a subsequent _mega pass_ is executed.

Reply:
The `LambdaShorthandToLambda` is defined as _being invalidated_ by `OperatorToFunction` pass as it is known that `LambdaShorthandToLambda` **needs to see result** of `OperatorToFunction` in its `prepare` method - that's only possible if the whole tree produced by `OperatorToFunction` is fed into `LambdaShorthandToLambda` - e.g. **these two mini passes cannot be merged**.